rtrlib/rtr/packets.c

```diff
@@ -481,6 +388,26 @@ static bool rtr_pdu_check_size(const struct pdu_header *pdu)
 		if (sizeof(struct pdu_reset_query) == pdu->len)
 			retval = true;
 		break;
+	case ASPA:
+		aspa_pdu = (const struct pdu_aspa *)pdu;
+		expected_size = sizeof(struct pdu_aspa);
+		if (aspa_pdu->len < expected_size) {
+			RTR_DBG1("PDU is too small to contain ASPA PDU!");
+			break;
+		}
+
+		// Check if the PDU really contains the ASPA PDU
+		uint16_t asn_count = ntohs(aspa_pdu->provider_count);
+
+		// ASN is 4 bytes each
+		expected_size += asn_count * sizeof(aspa_pdu->provider_asns[0]);
+		if (aspa_pdu->len != expected_size) {
+			RTR_DBG1("PDU is too small to contain valid ASPA PDU!");
```

Comment:
@fho if you look at the line above (L403) we calculate the expected size based on  the partially parsed telegram (`expected_size += asn_count * sizeof(aspa_pdu[0]->provider_asns[0])`). 

I changed the message to `The PDU is malformed, because the specified size doesn't match the real PDU size.`